frontend/src/libs/hooks/use-message-reducer/use-message-reducer.hook.ts

```diff
@@ -0,0 +1,77 @@
+import { ChatRole } from '#libs/enums/enums.js';
+import { useCallback, useReducer } from '#libs/hooks/hooks.js';
+import { type Message } from '#libs/types/types.js';
+
+type MessageReducer = {
+  messages: Message[];
+  addMessage: (text: string) => void;
+  newMessage: (message: Message) => void;
+};
+
+const LAST = -1;
+const FIRST = 0;
+
+type Action =
+  | { type: 'NEW_MESSAGE'; payload: Message }
+  | { type: 'ADD_MESSAGE'; payload: string };
+
+const messageReducer = (state: Message[], action: Action): Message[] => {
```

Comment:
As we are using async actions so local reducer is not enough for us. Create async action but not include api calls

Reply:
Can I use createAsyncThunk of Redux?

Reply:
You are right

Reply:
Solved!